### ISSUE TITLE
Animations start on the wrong frame

### ISSUE THREAD

**Issue by WAUthethird** [thargor6/mb3d#16]
Having an issue with MB3D (latest) where _all_ animations start their render position a few frames after the set start.

Below is an image I rendered of the actual first frame of an example animation. It is the first frame _I_ set for the animation. (The camera is moving left in this example)
![test](https://user-images.githubusercontent.com/17788745/109914587-8fc0f880-7c6d-11eb-8e2d-2e4716eea40b.png)

And here is the first image it output after starting the animation render.
![new000001](https://user-images.githubusercontent.com/17788745/109914685-c139c400-7c6d-11eb-903b-d345a21cf7c2.png)

This problem seems to occur all the way back to v1.99.0, and I would assume 1.98 as well.
It is very irritating, as my workflow requires very very precise timing in regards to frames - I sync to music.

**Comment by bezo97** [comment]
I'm pretty sure this bug was present in a lot earlier versions too, way before the 1.9x releases.

**Comment by thargor6** [comment]
I'm also pretty sure that I did not change there anything during the last years. 
Unfortunately, I could not even duplicate the problem. Added two keyframes and it worked as expected. What else is requiered?

**Comment by WAUthethird** [comment]
That's it, really. I changed nothing from the default aside from the image type.
You're sure it worked as expected? As in, you rendered the first frame both as a standalone render and as an animation to compare?

**Comment by WAUthethird** [comment]
Here's a test .m3a that exhibits the problem.
[test.zip](https://github.com/thargor6/mb3d/files/6087028/test.zip)

**Comment by thargor6** [comment]
I rendered the whole animation as single frames (which btw. did never stop, might this be a bug, too?) Which setting exactly would be "as animation"? 
Thanks for the file, will check that later.

**Comment by WAUthethird** [comment]
Yeah, I rendered as single frames as well. What I'm referring to is a render in the main program (i.e. sending the first position to the main program) versus an animation via the animations window. The never stopping issue is another problem; it does eventually stop but only after quite a few extra frames past the supposed stop number.

**Comment by bezo97** [comment]
I was able to replicate it, it's easier to see on a formula that's supposed to be symmetrical in the very first frame, such as Menger3:
![image](https://user-images.githubusercontent.com/26039350/110102581-3fe54d00-7da5-11eb-8bf5-826e34b5aba6.png)

**Comment by WAUthethird** [comment]
@thargor6 No pressure, but do you have any updates regarding this? I have a slightly time-sensitive project I'd really like to get going on.

**Comment by thargor6** [comment]
I'm currenty very busy with other projects and hoping for a free time-slot soon...

**Comment by thargor6** [comment]
could be then combined with the feature request regarding animation frames

**Comment by WAUthethird** [comment]
Hi again. I really don't want to come off as rude (I know you're doing this for free) but I'm just wondering if you have any updates on this?

**Comment by thargor6** [comment]
So I had a look at your example file. 
The reason seems to be Bezier interpolation (a Bezier curve does not exactly hit the interpolation points). 
There seem at least three workarounds:
a) use linear interpolation
b) insert a keyframe with the same position before the first keyframe
c) start with the camera a little bit more at the left
Does this make sense to you?

**Comment by WAUthethird** [comment]
Yes, that makes sense.
I tested linear interpolation and inserting a keyframe with the same position before the first keyframe.
Linear interpolation exhibits none of the problems that Bezier interpolation does, however this comes at the cost of clean movement, so for me at least, it is not an option.
Inserting a keyframe with the same position data as the first before the first keyframe does work provided I set the number of frames calculated to 1 (don't want it to render the same frame 50 times) but it does work.

However, there is another issue with the program that suggests this is a bug. On every animation render I do, the program usually keeps going past the set framelimit for a while before stopping. This means that camera movement is slower than accounted for, which makes syncing to music very difficult.
I have a suspicion that these two are connected, as the fact that the counter keeps going up (i.e. it'll say "584 out of 500 done") tells me there's something going wrong.

Perhaps this is related?
https://github.com/thargor6/mb3d/blob/c87fcf0c6b18be6cbdd2d9d3aae27e4050cd6957/Interpolation.pas#L638

**Comment by thargor6** [comment]
I see, and also faced this problem a long time ago. But looking at this code, I have to conclude that it would be easier to rewrite it than to understand it with all its hard coded values and many commented out parts. It seems that the original author also tried many things and then stopped somewhere. 
Maybe you could switch to Mandelbulber?

**Comment by WAUthethird** [comment]
Yeah. That definitely makes sense.
Okay, I'll probably try to use your workarounds for a while before trying to get myself to finally learn Mandelbulber haha

Thanks for the help!
